src/youtube/js/youtube.js

```diff
@@ -0,0 +1,374 @@
+        element.controller.notify(element.controller.EVENTS.ERROR, { "errorcode" : code });
+        // The owner of the requested video does not allow it to be played in embedded players. 
+        break;
+      case 150:
+        element.controller.notify(element.controller.EVENTS.ERROR, { "errorcode" : code });
+        // This error is the same as 101. It's just a 101 error in disguise! 
+        break;
+      }
+  };
+
+  /**
+   * @class OoyalaYoutubeVideoWrapper
+   * @classdesc Player object that wraps the video element.
+   * @property {object} controller A reference to the Ooyala Video Tech Controller
+   * @property {boolean} disableNativeSeek When true, the plugin should supress or undo seeks that come from
+   *                                       native video controls
+   */  
+  var OoyalaYoutubeVideoWrapper = function() {
+    this.controller = {};
+    this.disableNativeSeek = false;
+    var timeUpdateInterval = null;
+    var hasPlayed = false;
+    var videoEnded = false;
+
+    /**
+     * Triggers playback on the video element.
+     * @public
+     * @method OoyalaYoutubeVideoWrapper#play
+     */
+    this.play = function() {
+      if(playerReady)
+      {
+        youtubePlayer.playVideo();
+        this.controller.notify(this.controller.EVENTS.PLAY, { url: youtubeID });
+        updateTimerDisplay();
+        hasPlayed = true;
+      }
+      else {
+        javascriptCommandQueue.push(["play", null]);
+        playerNotready = false;
+      }
+    };
+
+    /**
+     * Triggers a pause on the video element.
+     * @public
+     * @method OoyalaYoutubeVideoWrapper#pause
+     */
+    this.pause = function(){
+      youtubePlayer.pauseVideo();
+      clearInterval(timeUpdateInterval);
+      timeUpdateInterval = null;
+    };
+    
+    /**
+     * Triggers a seek on the video element.
+     * @public
+     * @method OoyalaYoutubeVideoWrapper#seek
+     * @param {number} time The time to seek the video to (in seconds)
+     */
+    this.seek = function(time) {
+      if(playerReady)
+      {
+        youtubePlayer.seekTo(time,true);
+        this.controller.notify( this.controller.EVENTS.SEEKED);
+      }
+      else
+      {
+        javascriptCommandQueue.push(["seek", time]);
+      }
+    };
+
+    /**
+     * Triggers a volume change on the video element.
+     * @public
+     * @method OoyalaYoutubeVideoWrapper#setVolume
+     * @param {number} volume A number between 0 and 1 indicating the desired volume percentage
+     */
+    this.setVolume = function(volume) {
+      youtubePlayer.setVolume(volume*100);
+      this.controller.notify( this.controller.EVENTS.VOLUME_CHANGE, { "volume" :volume});
+    };
+
+    /**
+     * Sets the initial time of the video playback.
+     * @public
+     * @method OoyalaYoutubeVideoWrapper#setInitialTime
+     * @param {number} initialTime The initial time of the video (seconds)
+     */
+    this.setInitialTime = function(initialTime) {
+      if (!hasPlayed || videoEnded) {
```

Comment:
If the video has ended, do we care about the initial time anymore? Or does this function get passed 0 on replays.